contracts/NFTVault.sol

```diff
@@ -229,7 +235,7 @@ contract NFTVault is IERC721Receiver, IERC1155Receiver, AccessControl {
         bytes calldata data
     ) public virtual override returns (bytes4) {
         if (whiteList[from]) {
-            if (keys.transfer(from, TOKEN_AMOUNT)) {
+            if (keys.transfer(from, TOKEN_AMOUNT * BASE_UNIT)) {
```

Comment:
Wouldn be better to have `TOKEN_AMOUNT` already multiplied by BASE_UNIT on its definition?